src/CsvHelper/CsvReader.cs

```diff
@@ -90,6 +90,18 @@ public virtual bool ReadHeader()
 				throw new ReaderException(context, "Configuration.HasHeaderRecord is false.");
 			}
 
+			if (context.ReaderConfiguration.IsHeaderComment)
+			{
+				bool foundHeader = context.Record[0].StartsWith("#");

```

Comment:
use `context.Record[0].StartsWith('#');`  (overload that takes chr instead of string) or `context.Record[0][0] == '#';` for better performance